### PR TITLE
Supporting eth_sign in Signer

### LINKED ISSUE #1310: Refactor SigningQueue
Items in SigningQueue should have some fields already filled in (there shouldn't be `Options` there).
Alternatively, Signer UI should add missing gasPrice / gas using `estimateGas` for instance.

### LINKED ISSUE #1419: Make eth_sign work
Signer doesn't support eth_sign

### PR DESCRIPTION
**Pull request by tomusdrw**
Closes #1419 
Closes #1310 

Yet to do:
- [x] UI changes
- [x] `eth_postSign` (?) - support async queries for sign (similar to `eth_postTransaction`)

### CODE REVIEW

**Comment by gavofyork** [comment]
would have been nicer if the UI integration could have been reviewed in the same PR as the code behind it.

File: rpc/src/v1/traits/eth.rs [hunk 1]
```diff
@@ -206,26 +206,31 @@ pub trait EthSigning: Sized + Send + Sync + 'static {
...
 	fn check_transaction(&self, _: Params) -> Result<Value, Error>;
 
 	/// Should be used to convert object to io delegate.
 	fn to_delegate(self) -> IoDelegate<Self> {
 		let mut delegate = IoDelegate::new(Arc::new(self));
 		delegate.add_method("eth_sign", EthSigning::sign);
+		delegate.add_method("eth_postSign", EthSigning::post_sign);
 		delegate.add_method("eth_sendTransaction", EthSigning::send_transaction);
 		delegate.add_method("eth_postTransaction", EthSigning::post_transaction);
 		delegate.add_method("eth_checkTransaction", EthSigning::check_transaction);
```
Comment (gavofyork) [inline rpc/src/v1/traits/eth.rs]: `checkRequest` instead of `checkTransaction`?

File: rpc/src/v1/traits/eth.rs [hunk 2]
```diff
@@ -206,26 +206,31 @@ pub trait EthSigning: Sized + Send + Sync + 'static {
...
-	/// Will return a transaction ID for later use with check_transaction. 
+	/// Will return a transaction ID for later use with check_transaction.
 	fn post_transaction(&self, _: Params) -> Result<Value, Error>;
 
 	/// Checks the progress of a previously posted transaction.
 	/// Should be given a valid send_transaction ID.
 	/// Returns the transaction hash, the zero hash (not yet available),
-	/// or an error. 
+	/// or an error.
 	fn check_transaction(&self, _: Params) -> Result<Value, Error>;
```
Comment (gavofyork) [inline rpc/src/v1/traits/eth.rs]: `check_request` instead of `check_transaction`?

File: rpc/src/v1/traits/eth.rs (hunk 1 above)
Comment (tomusdrw) [inline rpc/src/v1/traits/eth.rs]: Yup, I wasn't sure if it's ok to break backward compatibility (and I suppose one of your dapps was using this method).

**Comment by gavofyork** [comment]
fair enough (though another reason to have these parity/ethcore JS extensions in the same repo).

File: rpc/src/v1/traits/eth.rs (hunk 1 above)
Comment (gavofyork) [inline rpc/src/v1/traits/eth.rs]: fair enough (though another reason to have these parity/ethcore JS extensions in the same repo).

**Pull request merged**